lib/internal/errors.js

```diff
@@ -261,10 +261,16 @@ function uvException(ctx) {
     message += ` -> '${dest}'`;
   }
 
+  // Reducing the limit improves the performance significantly. We do not loose
+  // the stack frames due to the `captureStackTrace()` function that is called
+  // later.
+  const tmpLimit = Error.stackTraceLimit;
+  Error.stackTraceLimit = 0;
   // Pass the message to the constructor instead of setting it on the object
   // to make sure it is the same as the one created in C++
   // eslint-disable-next-line no-restricted-syntax
   const err = new Error(message);
+  Error.stackTraceLimit = tmpLimit;
```

Comment:
This seems like something V8 folks should be made aware of. Since in this case the lazily populated `.stack` isn't being accessed yet setting the `Error.stackTraceLimit` to `0` improves basic error creation performance.

I'd rather this be tracked as a V8 issue and fixed there than throwing `Error.stackTraceLimit = 0` all over the Node codebase.

Reply:
Yes, but the V8 issues mentioned above
 * [8451](https://bugs.chromium.org/p/v8/issues/detail?id=8451) – loading Error.stackTraceLimit could be done faster:
   > We load Error.stackTraceLimit every time we collect a stack trace. Maybe this is something we want to improve? See https://docs.google.com/document/d/1wwigvXVQ-mdLmSfNnLie2xVQGCt-6AKhP1OuyE38Vw8/edit
 * [8452](https://bugs.chromium.org/p/v8/issues/detail?id=8452) – Faster stack traces:
   >Stack traces are not used a lot in benchmarks, but often on real web sites for telemetry purposes. Some Node.js frameworks also use it to implement async stack traces. We could use better performance for stack trace collection in general

don't clearly state this specific case though.

Reply:
I opened another issue to properly address this: https://bugs.chromium.org/p/v8/issues/detail?id=8555